docs/accessing_transaction_field.rst

```diff
@@ -56,21 +73,53 @@ Operator
 :any:`Txn.local_num_byte_slices() <TxnObject.local_num_byte_slices>`             :code:`TealType.uint64`   3                Maximum local byte strings in app schema
-:any:`Txn.extra_program_pages() <TxnObject.extra_program_pages>`                 :code:`TealType.uint64`   4                Number of extra program pages for app
-:any:`Txn.application_args <TxnObject.application_args>`                         :code:`TealType.bytes[]`  2                Array of application arguments
 :any:`Txn.accounts <TxnObject.accounts>`                                         :code:`TealType.bytes[]`  2                Array of application accounts
 :any:`Txn.assets <TxnObject.assets>`                                             :code:`TealType.uint64[]` 3                Array of application assets
 :any:`Txn.applications <TxnObject.applications>`                                 :code:`TealType.uint64[]` 3                Array of applications
-:any:`InnerTxn.created_asset_id() <TxnObject.created_asset_id>`                  :code:`TealType.uint64`   5                The ID of the newly created asset in this transaction. This is only valid on inner transactions.
-:any:`InnerTxn.created_application_id() <TxnObject.created_application_id>`      :code:`TealType.uint64`   5                The ID of the newly created application in this transaction. This is only valid on inner transactions.
-:any:`InnerTxn.logs <TxnObject.logs>`                                            :code:`TealType.bytes[]`  5                Array of application logged items. This is only valid on inner transactions.
+:any:`Txn.clear_state_program() <TxnObject.clear_state_program>`                 :code:`TealType.bytes`    2
+:any:`Txn.extra_program_pages() <TxnObject.extra_program_pages>`                 :code:`TealType.uint64`   4                Number of extra program pages for app
+:any:`Txn.application_args <TxnObject.application_args>`                         :code:`TealType.bytes[]`  2                Array of application arguments
+:any:`Txn.created_application_id() <TxnObject.created_application_id>`           :code:`TealType.uint64`   5                The ID of the newly created application in this transaction. In v5, only valid on inner transactions. >= v6 works with top-level and inner transactions.
+:any:`Txn.logs <TxnObject.logs>`                                                 :code:`TealType.bytes[]`  5                Array of application logged items. In v5, only valid on inner transactions. >= v6 works with top-level and inner transactions.
+:any:`Txn.num_logs <TxnObject.num_logs>`                                         :code:`TealType.uint64`   5                Number of log messages emitted by an application call.
+:any:`Txn.last_log <TxnObject.last_log>`                                         :code:`TealType.bytes[]`  6                The last message emitted. Empty bytes if none were emitted. Application mode only.
```

Comment:
`num_logs` should not be here, since it's not directly exposed. Instead, you do `Txn.logs.length()`

And `Txn.last_log` should be formatted with parenthesis at the end, i.e. `Txn.last_log()`. The convention is single fields need parenthesis to be called and arrays do not, since `Txn.logs()[0]` looks a bit strange

Reply:
@jasonpaulos Thanks for catching `last_log` invocation mistake and the extraneous `num_logs` reference.

I attempted to fix both issues in https://github.com/algorand/pyteal/pull/204/commits/d7b5738a8dfc8dbdcb90fe943e464f934dd2e82a.

Aside:  I originally added `num_logs` because it appears in `txn.TxnField` enum listing.  What heuristic can I apply to better gauge which fields are exposed in a first class manner?

Reply:
Only the methods/properties on [`TxnObject`](https://github.com/algorand/pyteal/blob/9521f62e9c0f442fe4bc3890c3cc1791704dd42e/pyteal/ast/txn.py#L274) are exposed. Currently this is every `TxnField` except `first_valid_time` (since it's not yet implemented by the AVM) and those that start with `num_`.